src/fstests/fstests.jinja2

```diff
@@ -36,6 +38,50 @@ class Job(BaseJob):
         except Exception as e:
             print('Kernel build error:', e)
 
+    def _get_xml_from_vm(self, kdir):
+        try:
+            result = subprocess.run([f"debugfs -R \"dump ext4/results-4k/results.xml {kdir}/results.xml\" /home/kernelci/alex/fstests-poc/fstests/disks/vdg"], check=True, shell=True)
+            if result.returncode == 0:
+                print(f"XML file in the output directory.")
+                return 'pass'
+        except Exception as e:
+            print('XML not found.', e)
+            return 'fail'
+
+    def _parse_to_json(self, node, tree, name=None):
+        if 'result' in tree:
+            node['name'] = name
+            node['result'] = tree['result']
+        else:
+            if (not name == None): node['name'] = name
+            child_nodes = []
+            for name in tree:
+                child_nodes.append(self._parse_to_json({}, tree[name], name))
+            node['child_nodes'] = child_nodes
+        return node
+
+    def _parse_results(self, kdir):
+        xml_file = ET.parse(f"{kdir}/results.xml")
+        root = xml_file.getroot()
+        tree = {}
+        for test_case in root.iter('testcase'):
+            if test_case.find('skipped') != None:
+                result = None
+            elif test_case.find('failure') != None:
+                result = 'fail'
+            else:
+                result = 'pass'
+            name = test_case.attrib['name'].split('/')
+            if not name[0] == 'ext4':
+                name.insert(0, 'ext4')
+            current = tree
+            for item in name:
+                if not item in current:
+                    current[item] = {}
+                current = current[item]
+            current['result'] = result
+        return self._parse_to_json({}, tree, 'FSTest Parsing Results')
```

Comment:
This should be `fstests` to match what is in the top-level node which was already sent by the `runner.py` script.
```suggestion
        return self._parse_to_json({}, tree, 'fstests')
```

Reply:
updated, now the root tree is fstests.